static/javascript.js

```diff
@@ -98,6 +103,11 @@
             validateButton.hide();
             // Hide loader icon.
             spinner.hide();
+            // Remove autoresponder error if successful validation.
+            var existingAutoresponder = document.querySelector('.smaily-settings').querySelector('.error-autoresponders');
+            if (existingAutoresponder !== null) {
+              document.querySelector('.smaily-settings').querySelector('.error-autoresponders').remove();
+            }
```

Comment:
It is not a good idea to remove the whole `.error-autoresponders` element.

Picture this scenario:
- a user successfully validates the API credentials;
- a success message is shown, but also the `.error-autoresponders` element is removed from DOM;
- the user now changes the credentials, making them invalid;
- tries to validate the changed credentials;
- error is not shown;

Use the same logic as in `displayMessage` function - remove the messages inside, but leave the container intact.

Reply:
Currently `.error-autoresponders` is on the same level as `.error `and `.notice `. In this case, should we separate `.error-autoresponders` into a new div element in the template?

Current layout:
![Smaiy plugin settings](https://i.imgur.com/WrBE5aj.png)

Reply:
Ah, good catch. I did not notice that the HTML for autoresponders error is different.

Perhaps it would be better to rework autoresponders error completely, then autoresponders error clean up could be removed entirely.

Make necessary changes in `templates/smaily-woocommerce-admin.php` to code:

```php
<div class="message-display"></div>
<?php if ( $autoresponder_error ) : ?>
<div class="error-autoresponders error notice">
    <p>
    <?php
        esc_html_e(
            'There seems to be a problem with your connection to Smaily. Please revalidate your credentials!',
            'smaily'
        );
    ?>
    </p>
</div>
```

for it to fill following conditions:
- when credentials are validated, the message is removed by functionality added to `displayMessage` by this PR;
- the message is an error;
- the message is dismissible;